### PR TITLE
Add page describing CVEs and their impact on the FIPS provider.

### PR DESCRIPTION
**Pull request by paulidale**
Note: there is no automation here.

### CODE REVIEW

File: news/fips-cve.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+and resolved.  While the majority of these vulnerabilities are unrelated
+to the validated FIPS provider, a few of them are applicable.  This table
+lists all of the CVEs issued since the FIPS provider's release and their
+relevance to it:
+
+<!--
+  -- Pandoc allocates horizontal space equally, we don't want that so override
+  -- This, however, removes spacing between table columns so we force some
+  -- using non-breaking spaces.
+  -->
```
Comment (levitte) [inline news/fips-cve.md]: Actually, it bases the size proportions on the table header underlining (i.e. line 24 below).

When I tried expanding the first underlining (after removing the style), i.e. to this:

```
------------- | :-:   | :-: | :--
```

I got a different spacing:

```
<colgroup>
<col style="width: 59%" />
<col style="width: 13%" />
<col style="width: 13%" />
<col style="width: 13%" />
</colgroup>
```

Comment (paulidale) [reply news/fips-cve.md]: Thanks, I wasn't aware of this and didn't find any documentation explaining it.

**Review by hlandau — APPROVED** [review]

File: news/fips-cve.md (hunk 1 above)
Comment (levitte) [reply news/fips-cve.md]: I just looked through the user guide, and indeed, there isn't much about it in there (and what is there is incredibly vague)

**Review by levitte — APPROVED** [review]

File: news/fips-cve.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+CVE-2023-0215 | 3.0.8 | no |
+CVE-2022-4450 | 3.0.8 | no |
+CVE-2022-4304 | 3.0.8 | yes | Timing side channel in RSA
+CVE-2022-4203 | 3.0.8 | no |
+CVE-2022-3996 | 3.0.8 | no |
+CVE-2022-3786 | 3.0.7 | no |
+CVE-2022-3602 | 3.0.7 | no |
+CVE-2022-3358 | 3.0.6 | no |
+CVE-2022-2274 | 3.0.5 | no | Bug introduced in 3.0.4 which isn't validated
+CVE-2022-2097 | 3.0.5 | no | Archicture (x86) is not part of validation
```
Comment (t8m) [inline news/fips-cve.md]: Typo: Architecture

File: news/fips-cve.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+CVE-2022-2068 | 3.0.4 | no |
+CVE-2022-1473 | 3.0.3 | no |
+CVE-2022-1434 | 3.0.3 | no |
+CVE-2022-1343 | 3.0.3 | no |
+CVE-2022-1292 | 3.0.3 | no |
+CVE-2022-0778 | 3.0.2 | maybe | Difficult to encounter inside FIPS boundary
+CVE-2021-4160 | 3.0.1 | no | Archicture (MIPS) is not part of validation
```
Comment (t8m) [inline news/fips-cve.md]: Typo: Architecture

File: news/fips-cve.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# CVEs and the FIPS provider
+
+After the release of OpenSSL 3.0.0, several CVEs have been identified
+and resolved.  While the majority of these vulnerabilities are unrelated
+to the validated FIPS provider, a few of them are applicable.  This table
+lists all of the CVEs issued since the FIPS provider's release and their
+relevance to it:
+
+
+CVE ID | Fixed | Impacted | Notes
```
Comment (t8m) [inline news/fips-cve.md]: Perhaps use `FIPS Provider Impact` instead of just `Impacted`

File: news/index.md [hunk 5]
```diff
@@ -12,5 +12,8 @@ If you think you have found a security bug, or want to look at all the
 vulnerabilities we have published and fixed, visit the
 [Vulnerabilities](vulnerabilities.html) page.
 
+For a list of CVEs and their impact on the FIPS provider, visit the
+[CVEs and FIPS](fips-cve.html) page.
+
```
Comment (t8m) [inline news/index.md:17]: Could you also link it from the Downloads page https://www.openssl.org/source/ ? There is a lot of text about FIPS provider in there so it seems good place to add this link as well.

**Review by t8m — CHANGES_REQUESTED** [review]
Good work!

File: news/fips-cve.md (hunk 4 above)
Comment (paulidale) [reply news/fips-cve.md]: I don't want this one being too long.  `FIPS?` might be better.
Once we've multiple FIPS providers validated, I'd like to list versions here.  It's pointless doing so with only one.

File: news/index.md (hunk 5 above)
Comment (paulidale) [reply news/index.md:17]: Added the link and reformatted the list of validated versions into a table.

**Comment by paulidale** [comment]
ping @hlandau @levitte

**Review by t8m — APPROVED** [review]

File: source/index.md [hunk 6]
```diff
@@ -24,10 +24,18 @@ to 3.1 or 3.0 as soon as possible. Extended support for 1.0.2 to gain access to
 security fixes for that version is [available](/support/contracts.html).
 
 
-The following OpenSSL version(s) are FIPS validated.
+The following OpenSSL version(s) are FIPS validated:
```
Comment (mattcaswell) [inline source/index.md:27]: Should we say somewhere here that you should use the latest OpenSSL version for the libraries in conjunction with the validated version for the FIPS provider?

File: news/fips-cve.md (hunk 4 above)
Comment (levitte) [reply news/fips-cve.md]: This whole file is about the FIPS provider...  not sure I understand what's unclear with "Impacted" in that case.  But of well, it was changed.

File: source/index.md [hunk 7]
```diff
@@ -24,10 +24,18 @@ to 3.1 or 3.0 as soon as possible. Extended support for 1.0.2 to gain access to
...
 
-OpenSSL 3.0.0. Refer to the [Certificate](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4282)
-and [Security Policy](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf)
+&nbsp;OpenSSL Version&nbsp; | &nbsp;Certificate&nbsp; | &nbsp;Security Policy&nbsp;
+:-: | :-: | :-:
+3.0.0 | [certificate](cert 3.0.0) | [security policy](secpol 3.0.0)
```
Comment (levitte) [inline source/index.md]: Those reference-style links are incorrect.  Here's what that line should look like:

```suggestion
3.0.0 | [certificate][cert 3.0.0] | [security policy][secpol 3.0.0]
```

Ref: https://www.markdownguide.org/basic-syntax/#reference-style-links

File: source/index.md [hunk 8]
```diff
@@ -24,10 +24,18 @@ to 3.1 or 3.0 as soon as possible. Extended support for 1.0.2 to gain access to
...
+
+[cert 3.0.0]: (https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4282)
+[secpol 3.0.0]: (https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf)
```
Comment (levitte) [inline source/index.md]: No parentheses around the URLs in reference style links.  In other words:

```suggestion
[cert 3.0.0]: https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4282
[secpol 3.0.0]: https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4282.pdf
```

**Review by levitte — CHANGES_REQUESTED** [review]

File: source/index.md (hunk 6 above)
Comment (levitte) [reply source/index.md:27]: Side note: I find it a bit weird that we're talking about validation of the whole of OpenSSL...  it's really only the FIPS provider (including the parts of OpenSSL that it uses, of course) that's validated, right?

File: source/index.md (hunk 7 above)
Comment (paulidale) [reply source/index.md]: Fixed

File: source/index.md (hunk 6 above)
Comment (paulidale) [reply source/index.md:27]: This is mentioned just after the table:

> Other OpenSSL Releases MAY use the validated FIPS provider, but MUST NOT build and use their own FIPS provider. For example you can build OpenSSL 3.1 and use the OpenSSL 3.0.0 FIPS provider with it.

It's not as forceful as what you are suggesting.

**Review by hlandau — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**